js/ui/appDisplay.js

```diff
@@ -261,8 +261,18 @@ class BaseAppView {
 
     moveItem(item, newPosition) {
         let visibleItems = this._allItems.filter(item => item.actor.visible);
+
+        // Avoid overflow, this happens when trying to move an icon behind the
+        // AppCenterIcon
+        if (newPosition >= visibleItems.length)
+            return -1;
+
         let targetId = visibleItems[newPosition].id;
 
+        // Do not allow to move icons before the HackAppIcon
+        if (targetId === Clubhouse.getClubhouseApp().get_id())
+            return -1;
```

Comment:
Hm, I think this is the wrong place to put this check. One problem with the check being here is that there would be a visual feedback for dropping the icon (the Hack icon would "nudge"), but dropping would be ignored. Perhaps adding a similar check in `AppDisplay.handleDragOver()`.

Reply:
Thank you, I've done that in the `handleDragOver` in the `HackAppIcon` class and it's better to have the code in this class instead of the parent class.